cmake/common/px4_base.cmake

```diff
@@ -1083,11 +1099,25 @@ function(px4_generate_parameters_source)
 	set_source_files_properties(${generated_files}
 		PROPERTIES GENERATED TRUE)
 	px4_join(OUT module_list  LIST ${MODULES} GLUE ",")
-	add_custom_command(OUTPUT ${generated_files}
-		COMMAND ${PYTHON_EXECUTABLE} ${PX4_SOURCE_DIR}/Tools/px_generate_params.py
-		--xml ${XML} --modules ${module_list} --dest ${CMAKE_CURRENT_BINARY_DIR}
-		DEPENDS ${XML} ${DEPS}
+
+	if("${BOARD}" STREQUAL "eagle" OR "${BOARD}" STREQUAL "excelsior")
+		# If the board is a Snapdragon Flight "Eagle" or "Excelsior" board
+		# then include all parameters defined by all modules.  This is required
+		# for parameter synchronization between the ARM and DSP processors.
+		message("Generating parameter source files for board: ${BOARD}")
+		add_custom_command(OUTPUT ${generated_files}
+			COMMAND ${PYTHON_EXECUTABLE} ${PX4_SOURCE_DIR}/Tools/px_generate_params.py
+				--xml ${XML} --dest ${CMAKE_CURRENT_BINARY_DIR}
+			DEPENDS ${XML} ${DEPS}
+		)
+	else()
+		add_custom_command(OUTPUT ${generated_files}
```

Comment:
All good points.  I was trying to minimize the number of lines I touched and wasn't thinking from this perspective.  I will make the changes later today.